### DISCUSSION TITLE
Lazy loading vimspector

### DISCUSSION THREAD

**Discussion by OmriLevy888** [wbthomason/packer.nvim#531] [Q&A] [answered]
New to packer.nvim, so this is most likely a newbie question.

I cloned NvChad and started configuring it (the call to packer.setup is in pluginList.lua). I added vimspector as such:
```lua
use {
    "puremourning/vimspector",
    cmd = {
        "call vimspector#Launch()"
    }
}
```

I tried to use PackerStatus to see if vimspector is loaded after installing it and saw that it was loaded, even though I expected it to be loaded only once the Launch function is invoked. 

I have vimspector#Launch bound to start a new debugging session, so I would like it to be the trigger to load vimspector.

P.S. there is also vimspector#Reset to unload vimspector. Just out of curiosity, does packer support unloading a package and then reload it again using some trigger (such as the cmd trigger)? I guess this would cause wasting too much time on loading and unloading but it could be a neat way to use a lot of plugins on weaker machines where the concern is not only startup time but also RAM and such.

Thanks for any help!

**Comment by wbthomason** [comment, marked as answer]
Welcome! I have a few comments:
- First of all, did you run `PackerSync` after saving your changes? If `vimspector` was originally a `start` plugin, you need this (or `PackerCompile` + `PackerUpdate` independently) to build the lazy-loader and move `vimspector` to the `opt` plugin dir.
- Second, I would suggest using the `fn` key instead of `cmd`, as in `fn = 'vimspector#Launch'`. This more closely matches your desired load conditions, and I do not believe that the command you've specified would work anyway (`cmd` is for command names rather than commands with specific arguments).
- Finally, we do not support unloading plugins - we'd need some way to "un-source" Vimscript files, which I'm not aware of any way to do, as well as some way to make `packadd` properly source a plugin again.

I hope this helps!

**Reply by OmriLevy888** [reply]
I see! This works now, thanks for the clarification, quite amazed by how flexible this is.
